src/range_proof.rs

```diff
@@ -600,6 +601,72 @@ impl RangeProof {
         Ok(masks)
     }
 
+    /// Serializes the proof into a byte array of 32-byte elements
+    pub fn to_bytes(&self) -> Vec<u8> {
+        // 6 elements, 2 vectors
+        let mut buf = Vec::with_capacity((6 + self.li.len() + self.ri.len()) * 32);
+        for l in &self.li {
+            buf.extend_from_slice(l.as_bytes());
+        }
+        for r in &self.ri {
+            buf.extend_from_slice(r.as_bytes());
+        }
+        buf.extend_from_slice(self.a.as_bytes());
+        buf.extend_from_slice(self.a1.as_bytes());
+        buf.extend_from_slice(self.b.as_bytes());
+        buf.extend_from_slice(self.r1.as_bytes());
+        buf.extend_from_slice(self.s1.as_bytes());
+        buf.extend_from_slice(self.d1.as_bytes());
+        buf
+    }
+
+    /// Deserializes the proof from a byte slice
+    pub fn from_bytes(slice: &[u8]) -> Result<RangeProof, ProofError> {
+        if slice.len() % 32 != 0 {
+            return Err(ProofError::InvalidLength(
+                "Serialized proof bytes must be a factor of 32".to_string(),
+            ));
+        }
+        let num_elements = slice.len() / 32;
+        if (num_elements - 6) % 2 != 0 {
+            return Err(ProofError::InvalidLength(
+                "Serialized proof has incorrect number of elements".to_string(),
+            ));
+        }
+        let n = (num_elements - 6) / 2;
```

Comment:
For `slice.len() == 0` this wraps around to a very large number.

Reply:
Thanks, fixed and tested